types/src/params.rs

```diff
@@ -350,7 +350,7 @@ impl<'a> SubscriptionId<'a> {
 }
 
 /// Request Id
-#[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize)]
+#[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize, PartialOrd, Ord)]
 #[serde(deny_unknown_fields)]
 #[serde(untagged)]
 pub enum Id<'a> {
```

Comment:
I didn't notice offhand; is a borrowed version of an ID ever used here? I pretty much assume every new ID will need to be an owned variant anyway. so I wonder how useful the lifetime/`Cow` is?

(thinking down that line, in general any `serde_json::Value` _could_ be returned as an ID (whether or not certain values make sense, who knows) so I was wondering whether using that here would simplify anything)

Reply:
I think you are right at least in the point of view for the client.

However, I think we actually can use the lifetime and do zero-alloc deserialization in the servers with `#[serde](borrow)]`